lib/lock.js

```diff
@@ -170,8 +170,17 @@ Lock.prototype.extend = function(time, fn) {
   var key    = this._key;
   var shaval = this._shavaluator;
 
+  if (!fn && typeof time === 'function') {
+    fn = time;
```

Comment:
Makes sense! I updated the logic here: https://github.com/danielstjules/redislock/blob/master/lib/lock.js#L177-L179 Thanks again!